docs/development/visualize/development-using-visualizations.asciidoc

```diff
@@ -0,0 +1,103 @@
+
+`<visualize saved-id="'447d2930-9eb2-11e7-a956-5558df96e706'"></visualize>`
+
+For the above to work with time based visualizations time picker must be present (enabled) on the page. For scenarios
+where timepicker is not available time range can be passed in as additional parameter:
+
+`<visualize saved-id="'447d2930-9eb2-11e7-a956-5558df96e706'" 
+time-range="{ max: '2017-09-21T21:59:59.999Z', min: '2017-09-18T22:00:00.000Z' }"></visualize>`
+
+Once <visualize> is done rendering the element will emit `renderComplete` event.
+
+When more control is required over the visualization you may prefer to load the saved object yourself and then pass it
+to `<visualize>`
+
+`<visualize saved-obj='savedVis' app-state='appState' ui-state='uiState' editor-mode='false'></visualize>` where
+
+`savedVis` is an instance of savedVisualization object, which can be retrieved using `savedVisualizations` service
+which is explained later in this documentation.
+
+`appState` is an instance of `AppState`. <visualize> is expecting two keys defined on AppState:
+
+- `filters` which is an instance of searchSource filter object and
+- `query` which is an instance of searchSource query object
+
+if not provided <visualize> will still work, just without monitoring the AppState.
+
+`uiState` should be an instance of `PersistedState`. if not provided visualize will generate one, 
+but you will have no access to it. Its used to store viewer specific information (like is legend toggled or not)
+
+`editor-mode` defines if <visualize> should render in editor or in view mode.
+
+*code example: showing saved visualization, without linking to querybar or filterbar*
+["source","html"]
+-----------
+<div ng-controller="KbnTestController" class="test_vis">
+  <visualize saved-obj='savedVis'></visualize>
+</div>
+-----------
+["source","js"]
+-----------
+import { uiModules } from 'ui/modules';
+
+uiModules.get('kibana')
+.controller('KbnTestController', function ($scope, AppState, savedVisualizations) {
+  const visId = 'enter_your_vis_id';
+  savedVisualizations.get(visId).then(savedVis => $scope.savedObj = savedVis);
+});
+-----------
+
+When <visualize> is ready it will emit `ready:vis` event on the root scope.
+When <visualize> is done rendering it will emit `renderComplete` event on the element.
+
+==== `<visualization>` directive
+`<visualization>` directive takes visualization config and data and renders the appropriate visualization.
+
+`<visualization vis='vis' vis-data='visData' ui-state='uiState' ></visualization>` where
+
+`vis` is an instance of `Vis` object. The constructor takes 3 parameters:
+
+- indexPattern (string): the indexPattern you want to pass to the visualization
+- visConfig (object): the configuration object
+- uiState (object): uiState object you want to pass to Vis. If not provided Vis will create its own.
+
+`visData` is the data object. Each visualization defines a `responseHandler`, which defines the format of this object.
```

Comment:
consider:

> `visData` is the data object visualization expects. At the moment, this object is still very much coupled to the Elasticsearch response. We foresee this will change as the Kibana codebase evolves.